src/coprocessor/endpoint.rs

```diff
@@ -301,153 +427,236 @@ impl ReqContext {
         }
         Ok(())
     }
+
+    pub fn renew_streaming_outdated(&mut self) {
+        let now = Instant::now_coarse();
+        if !self.is_stream_head || self.deadline <= now {
+            self.is_stream_head = false;
+            self.deadline = now + Duration::from_secs(REQUEST_MAX_STREAM_HANDLE_SECS);
+        }
+    }
 }
 
+#[derive(Debug)]
+struct RequestTracker {
```

Comment:
What is `Tracker`?

Reply:
It's a lifetime tracker for every DAG request. After we finish the process of a request, we will call `RequestTracker::drop()` to release some resources.